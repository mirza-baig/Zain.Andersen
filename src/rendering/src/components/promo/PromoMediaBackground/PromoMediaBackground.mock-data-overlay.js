const defaultData = {
    rendering: {
      dataSource: 'sampledatasource',
      componentName: 'PromoMediaBackground',
    },
    fields: {
        "overlay": {
            "id": "cec0af91-e32d-4183-86b2-1594757d8052",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Overlay/Black",
            "name": "Black",
            "displayName": "Black",
            "fields": {
                "Value": {
                    "value": "black"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "componentSpacing": null,
        "sectionId": {
            "value": ""
        },
        "eyebrowLevel": {
            "id": "7f10288a-beb1-4d95-a139-32df40094441",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3",
            "name": "Heading 3",
            "displayName": "Heading 3",
            "fields": {
                "Value": {
                    "value": "h3"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "eyebrowText": {
            "value": "Eyebrow"
        },
        "headlineText": {
            "value": "Promo Heading"
        },
        "headlineLevel": {
            "id": "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
            "name": "Heading 2",
            "displayName": "Heading 2",
            "fields": {
                "Value": {
                    "value": "h2"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "body": {
            "value": "<strong>Lorem Ipsum</strong>&nbsp;is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. [No text in field]"
        },
        "fontColor": {
            "id": "12a26c7b-7f96-4c40-9762-187d4132dd6b",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Font-Color/White",
            "name": "White",
            "displayName": "White",
            "fields": {
                "Value": {
                    "value": "white"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "cta1Modal": null,
        "cta1Style": {
            "id": "49a23327-0397-4cce-a930-e76918d37c42",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary",
            "name": "Primary",
            "displayName": "Primary",
            "fields": {
                "Value": {
                    "value": "primary"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "cta1Icon": {
            "id": "50590edc-7ea7-4436-9a3e-701c87a07db2",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow",
            "name": "Arrow",
            "displayName": "Arrow",
            "fields": {
                "Value": {
                    "value": "arrow"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "cta1Link": {
            "value": {
                "href": "/",
                "text": "Default State",
                "anchor": "",
                "linktype": "internal",
                "class": "",
                "title": "",
                "target": "",
                "querystring": "",
                "id": "{7FB335D2-8E99-458E-9EF9-562A78CCB821}"
            }
        },
        "cta1ModalLinkText": {
            "value": ""
        },
        "primaryImage": {
            "value": {
                "src": "https://preview.local.renewalbyandersen.com/-/media/Project/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Images/RBA.png?h=260&iar=0&w=389&hash=5F55CA02F4BF7E2C5EB111D09D0E1132",
                "alt": "RBA",
                "width": "389",
                "height": "260"
            }
        },
        "primaryImageMobile": {
            "value": {}
          },
        "primaryImageMobileFocusArea": null,
        "primaryVideo": {
            "id": "25790bd9-b6c3-469a-a9a7-926810638542",
            "url": "/Data/Elements/Videos/YouTube-Video",
            "name": "YouTube Video",
            "displayName": "YouTube Video",
            "fields": {
                "youTubeAutoLoop": {
                    "value": true
                },
                "youTubeAutoPlay": {
                    "value": true
                },
                "youTubeDisableKeyboard": {
                    "value": false
                },
                "youTubeMute": {
                    "value": false
                },
                "youTubeShowControls": {
                    "value": false
                },
                "youTubeClosedCaptions": {
                    "value": false
                },
                "youTubeModestBranding": {
                    "value": false
                },
                "videoHeight": {
                    "value": 338.0
                },
                "videoId": {
                    "value": "K-eDe8GDMUo"
                },
                "videoLazyLoad": {
                    "value": false
                },
                "videoWidth": {
                    "value": 600.0
                }
            },
            "templateId": "10ffc85c-56a6-4a99-953c-56bb3405c1e6",
            "templateName": "YouTube Video"
        },
        "eventName": {
            "value": ""
        },
        "eventZone": {
            "value": ""
        },
        "eventType": {
            "value": ""
        }
    },};

export default defaultData;
