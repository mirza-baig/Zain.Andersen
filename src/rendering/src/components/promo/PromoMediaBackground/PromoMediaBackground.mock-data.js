const defaultData = {
    rendering: {
      dataSource: 'sampledatasource',
      componentName: 'PromoMediaBackground',
    },
    fields: {
        "overlay":  {
            "id": "5086f411-a48f-410f-8b24-580d39057406",
            "url": "http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Overlay/None",
            "name": "None",
            "displayName": "None",
            "fields": {
                "Value": {
                    "value": ""
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "componentSpacing": null,
        "sectionId": {
            "value": ""
        },
        "eyebrowLevel": {
            "id": "7f10288a-beb1-4d95-a139-32df40094441",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-3",
            "name": "Heading 3",
            "displayName": "Heading 3",
            "fields": {
                "Value": {
                    "value": "h3"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "eyebrowText": {
            "value": "Eyebrow"
        },
        "headlineText": {
            "value": "Promo Heading"
        },
        "headlineLevel": {
            "id": "0f556f3a-bbad-4aa5-952d-b79003b39cd6",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Headline-Levels/Heading-2",
            "name": "Heading 2",
            "displayName": "Heading 2",
            "fields": {
                "Value": {
                    "value": "h2"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "body": {
            "value": "<strong>Lorem Ipsum</strong>&nbsp;is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. [No text in field]"
        },
        "fontColor": {
            "id": "49b57385-1e72-494d-a630-e60cf0a01562",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Promo-Font-Color/Gray",
            "name": "Gray",
            "displayName": "Gray",
            "fields": {
                "Value": {
                    "value": "gray"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "cta1Modal": null,
        "cta1Style": {
            "id": "49a23327-0397-4cce-a930-e76918d37c42",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary",
            "name": "Primary",
            "displayName": "Primary",
            "fields": {
                "Value": {
                    "value": "primary"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "cta1Icon": {
            "id": "50590edc-7ea7-4436-9a3e-701c87a07db2",
            "url": "https://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow",
            "name": "Arrow",
            "displayName": "Arrow",
            "fields": {
                "Value": {
                    "value": "arrow"
                }
            },
            "templateId": "d2923fee-da4e-49be-830c-e27764dfa269",
            "templateName": "Enum"
        },
        "cta1Link": {
            "value": {
                "href": "/",
                "text": "Default State",
                "anchor": "",
                "linktype": "internal",
                "class": "",
                "title": "",
                "target": "",
                "querystring": "",
                "id": "{7FB335D2-8E99-458E-9EF9-562A78CCB821}"
            }
        },
        "cta1ModalLinkText": {
            "value": ""
        },
        "primaryImage": {
            "value": {
                "src": "https://preview.local.renewalbyandersen.com/-/media/Project/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Picture/photo-1556594472-e9b933db923c.jpg?h=900&iar=0&w=1200&hash=4507CA7F78A54822647815B8A2857B35",
                "alt": "jmage",
                "width": "1200",
                "height": "900"
            }
        },
        "primaryImageMobile": {
            "value": {
                "src": "https://preview.local.renewalbyandersen.com/-/media/Project/AndersenCorporation/RenewalByAndersen/RenewalByAndersen/Images/Portrait/PHTO-2013-Beauty-Andersen-Frenchwood-400-Series-Hinged-Patio-Door-20A-RGB-4.png?h=492&iar=0&w=491&hash=62C4297EC108672D90A8037CEFBEF3A3",
                "alt": "Image",
                "width": "491",
                "height": "492"
            }
        },
        "primaryImageMobileFocusArea": null,
        "primaryVideo": null,
        "eventName": {
            "value": ""
        },
        "eventZone": {
            "value": ""
        },
        "eventType": {
            "value": ""
        }
    },};

export default defaultData;
