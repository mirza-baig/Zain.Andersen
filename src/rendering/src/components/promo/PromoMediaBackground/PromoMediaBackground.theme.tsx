// Lib
import { ThemeFile } from 'lib/context/ThemeContext';

export const PromoMediaBackgroundTheme: ThemeFile = {
  aw: {
    classes: {
      /** Theme classes **/
      mediaContainer: 'relative max-md:mb-[15.75px]',
      overlay: 'absolute z-[1] w-full h-full bg-[rgba(0,0,0,0.5)] left-0 top-0',
      iconWrapper: 'mr-s mb-s md:mr-l md:mb-l',
      contentWrapper: 'max-md:px-m md:absolute w-full h-full z-[1] flex items-center md:top-0',
      contentContainer: ' md:ml-[8%] flex flex-col basis-full mx-auto',
      eyebrow: 'my-xxxs md:max-w-[585px] text-sm-xs md:text-xs font-medium uppercase font-sans',
      headline: 'my-xxxs md:max-w-[585px] text-sm-l leading-[2.35rem] lg:text-m font-heavy',
      bodyClass: 'my-xxxs  leading-[22px] font-regular md:max-w-[585px] md:text-xs md:font-medium',
      buttonGroupClass: {
        wrapper: 'my-s md:my-xs',
        cta1Classes: 'text-black',
      },
    },
  },
  rba: {
    classes: {
      /** Theme classes **/
      mediaContainer: 'relative max-md:mb-m',
      overlay: 'absolute z-[1] w-full h-full bg-[rgba(0,0,0,0.5)] left-0 top-0',
      iconWrapper: 'mr-s mb-s md:mr-ml md:mb-ml',
      contentWrapper:
        'col-span-12 max-md:px-m md:absolute w-full h-full z-[1] flex items-center md:top-0',
      contentContainer: 'col-span-12 md:ml-[8%] flex flex-col basis-full mx-auto',
      eyebrow:
        'md:my-xxxs md:max-w-[490px] font-semi-bold uppercase font-serif leading-[14px] text-sm',
      headline: 'mt-[5px] text-sm-s md:my-xxxs md:text-l md:max-w-[490px] font-medium',
      bodyClass:
        'mt-[5px] md:my-xxxs md:max-w-[490px] text-[0.850rem] leading-[1.35rem] text-dark-gray md:text-lg md:leading-6',
      buttonGroupClass: {
        wrapper: 'my-s md:my-xs',
        cta1Classes: 'md:bg-white md:text-black',
      },
    },
  },
};
/* Example of more advanced builder */
/*
import classNames from 'classnames';
const getDynamicStyles = (theme: ThemeName): string => {
  return classNames('text-xl', theme === 'aw' ? 'color-orange' : 'color-green');
};
export const PromoMediaBackgroundTheme = (): ThemeFile => {
  return {
    aw: {
      classes: {
        // Insert Theme classes here  - sample function call
        headline: classNames(getDynamicStyles('aw')),
      },
    },
    rba: {
      classes: {
        // Insert Theme classes here  - sample function call
        headline: classNames(getDynamicStyles('rba')),
      },
    },
  };
};
*/
