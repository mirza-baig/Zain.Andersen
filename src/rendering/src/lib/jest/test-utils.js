// Global
import { render, fireEvent, waitFor } from '@testing-library/react';
import { ComponentPropsContext } from '@sitecore-jss/sitecore-jss-nextjs';
import { FastSitecoreContext } from 'lib/overrides/FastSitecoreContext';

// Lib
import { createSampleComponentFactory } from 'lib/mocks/mock-placeholder';

export const eeSitecoreContext = {
  route: {
    name: 'Foo',
    placeholders: {},
  },
  context: {
    pageEditing: true,
    site: {
      name: 'JssTestWeb',
    },
    language: 'en',
  },
};

export const renderComponent = (
  Component,
  {
    componentProps = {},
    staticProps = {},
    componentFactory = createSampleComponentFactory(),
    sitecoreContext = {
      route: {
        name: 'Foo',
        placeholders: {},
      },
    },
  }
) => {
  const mockLayoutData = {
    sitecore: sitecoreContext,
  };

  return render(
    <ComponentPropsContext value={staticProps}>
      <FastSitecoreContext componentFactory={componentFactory} layoutData={mockLayoutData}>
        <Component {...componentProps} />
      </FastSitecoreContext>
    </ComponentPropsContext>
  );
};

export const snapshot = (Component, mockData) => {
  const component = renderComponent(Component, mockData);
  expect(component.container.firstChild).toMatchSnapshot();
  return component;
};

export const hasDataComponent = async (component, name) => {
  await waitFor(() => {
    const nodes = component.baseElement.querySelectorAll(`[data-component="${name}"]`);
    return expect(nodes.length).toBe(1);
  });
};

export const mockDateAs = (epochMs) => {
  if (typeof epochMs !== 'number') {
    console.error(
      'mockDateAs only accepts epoch values in milliseconds to prevent timezones from factoring in to Date creation.'
    );
  }

  jest.spyOn(global.Date, 'now').mockImplementation(() => new Date(epochMs).valueOf());
};

export const testFieldError = async (Component, mockData, { label, value, error }) => {
  const component = renderComponent(Component, mockData);
  const field = component.getByLabelText(label);
  fireEvent.change(field, { target: { value } });
  field.blur();
  await waitFor(() => {
    const err = component.queryByText(error);
    return expect(err).not.toBe(null);
  });
};
