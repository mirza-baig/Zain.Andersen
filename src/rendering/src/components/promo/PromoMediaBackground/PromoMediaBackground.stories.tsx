// Global
import { Story, Meta } from '@storybook/react';
// Local
import PromoMediaBackground, { PromoMediaBackgroundProps } from './PromoMediaBackground';
import defaultData from './PromoMediaBackground.mock-data';
import overlayData from './PromoMediaBackground.mock-data-overlay';

export default {
  title: 'Promo/PromoMediaBackground',
  component: PromoMediaBackground,
} as Meta;
const Template: Story<PromoMediaBackgroundProps> = (props) => <PromoMediaBackground {...props} />;
export const Default = Template.bind({});
Default.args = defaultData;
export const Overlay = Template.bind({});
Overlay.args = overlayData;
