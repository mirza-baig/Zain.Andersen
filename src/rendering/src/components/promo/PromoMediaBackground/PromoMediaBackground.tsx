// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
import { useState } from 'react';
import classNames from 'classnames';
// Components
import { PromoMediaBackgroundTheme } from './PromoMediaBackground.theme';
import { Headline } from 'src/helpers/Headline';
import RichTextWrapper from 'src/helpers/RichTextWrapper/RichTextWrapper';
import { ButtonGroup } from 'src/helpers/ButtonGroup';
import { Eyebrow } from 'src/helpers/Eyebrow';
import { MediaPrimary } from 'src/helpers/Media';
import { useCurrentScreenType } from 'lib/utils/get-screen-type';
import { getEnum } from 'lib/utils';
import { SvgIcon } from 'src/helpers/SvgIcon';

export type PromoMediaBackgroundProps =
  Feature.EnterpriseWeb.Enterprise.Components.Promo.PromoMediaBackground.PromoMediaBackground;
const PromoMediaBackground = (props: PromoMediaBackgroundProps) => {
  const { fields } = props;
  const { themeName, themeData } = useTheme(PromoMediaBackgroundTheme);
  const { screenType } = useCurrentScreenType();
  const [isVideoPaused, setIsVideoPaused] = useState(false);

  if (!fields) {
    return null;
  }

  const isDesktop = screenType !== 'sm' ? true : false;
  let hasOverlay = getEnum(fields.overlay) ? true : false;

  if (themeName === 'rba') {
    if (
      fields.headlineText.value.length > 0 ||
      fields.body.value.length > 0 ||
      fields.eyebrow.value.length > 0 ||
      fields.cta1Link.value.text.length > 0
    ) {
      hasOverlay = true;
    } else if (getEnum(fields.overlay) === '') {
      hasOverlay = false;
    }
  }

  let fontColorClass = '';

  if (hasOverlay) {
    fontColorClass = 'md:text-white';
  } else {
    switch (getEnum(fields.fontColor)) {
      case 'gray':
        fontColorClass = 'md:text-dark-gray';
        break;
      case 'white':
        fontColorClass = 'md:text-white';
        break;
      default:
        fontColorClass = 'md:text-black';
        break;
    }
  }

  const togglePlayPause = () => {
    const videoId = (
      props.fields?.primaryVideo as Feature.EnterpriseWeb.Enterprise.Elements.Media.YouTubeVideo
    )?.fields.videoId.value;

    const iframeContentWindow = (document.getElementById(videoId) as HTMLIFrameElement)
      ?.contentWindow;

    if (iframeContentWindow) {
      iframeContentWindow.postMessage(
        `{"event":"command", "func":"${isVideoPaused ? 'playVideo' : 'pauseVideo'}", "args":""}`,
        '*'
      );

      setIsVideoPaused(!isVideoPaused);
    }
  };
  return (
    <div data-component="promo/promomediabackground" className="relative m-auto max-w-[1440px]">
      <div className={themeData.classes.mediaContainer}>
        <MediaPrimary
          imageLayout="fill"
          ratio="hero"
          additionalDesktopClasses="h-[280px] md:h-[620px]"
          additionalMobileClasses="h-[280px]"
          {...props}
        />
        {fields.primaryVideo && (
          <div
            className={classNames(
              'absolute right-0 bottom-0 z-10 cursor-pointer rounded-full bg-black',
              themeData.classes.iconWrapper
            )}
            onClick={togglePlayPause}
          >
            {isVideoPaused ? <SvgIcon icon="play" size="lg" /> : <SvgIcon icon="pause" size="lg" />}
          </div>
        )}
      </div>
      {hasOverlay && isDesktop && <div className={themeData.classes.overlay}></div>}
      <div className={classNames(themeData.classes.contentWrapper, fontColorClass)}>
        <div className={classNames(themeData.classes.contentContainer, fontColorClass)}>
          <Eyebrow classes={classNames(themeData.classes.eyebrow, fontColorClass)} {...props} />
          <Headline classes={classNames(themeData.classes.headline, fontColorClass)} {...props} />
          <RichTextWrapper
            classes={classNames(themeData.classes.bodyClass, fontColorClass)}
            field={props.fields?.body}
          />
          {fields.cta1Link && fields.cta1Link.value.href && (
            <ButtonGroup classes={themeData.classes.buttonGroupClass} {...props} />
          )}
        </div>
      </div>
    </div>
  );
};

export default withDatasourceCheck()<PromoMediaBackgroundProps>(PromoMediaBackground);
