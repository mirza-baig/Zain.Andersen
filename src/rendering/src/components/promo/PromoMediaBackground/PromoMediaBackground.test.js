// Global
import dynamic from 'next/dynamic';
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Components
import IconNewTab from 'src/helpers/SvgIcon/icons/icon--new-tab';
// Local
import PromoMediaBackground from './PromoMediaBackground';
import defaultData from './PromoMediaBackground.mock-data';

// Mock out the SvgIcon dynamic import
jest.mock('next/dynamic');

beforeAll(() => {
  dynamic.mockImplementation(() => IconNewTab);
});
it('renders correctly', async () => {
  const component = snapshot(PromoMediaBackground, { componentProps: defaultData });
  await hasDataComponent(component, 'promo/promomediabackground');
});
