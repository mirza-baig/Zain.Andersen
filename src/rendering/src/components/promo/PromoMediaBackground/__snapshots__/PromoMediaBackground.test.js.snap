// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="relative m-auto max-w-[1440px]"
  data-component="promo/promomediabackground"
>
  <div
    class="relative max-md:mb-[15.75px]"
  >
    <div
      class="relative mx-auto hidden md:block aspect-hero   h-[280px] md:h-[620px]"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="jmage"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="relative block w-full md:hidden h-[280px]"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="Image"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
        />
        <noscript />
      </span>
    </div>
  </div>
  <div
    class="max-md:px-m md:absolute w-full h-full z-[1] flex items-center md:top-0 md:text-dark-gray"
  >
    <div
      class=" md:ml-[8%] flex flex-col basis-full mx-auto md:text-dark-gray"
    >
      <div
        class="my-xxxs md:max-w-[585px] text-sm-xs md:text-xs font-medium uppercase font-sans md:text-dark-gray"
      >
        <h3>
          Eyebrow
        </h3>
      </div>
      <div
        class="items-top font-sans [&_a:hover]:underline my-xxxs md:max-w-[585px] text-sm-l leading-[2.35rem] lg:text-m font-heavy md:text-dark-gray"
      >
        <h2>
          Promo Heading
        </h2>
      </div>
      <div
        class="body-copy my-xxxs  leading-[22px] font-regular md:max-w-[585px] md:text-xs md:font-medium md:text-dark-gray"
      >
        <div>
          <strong>
            Lorem Ipsum
          </strong>
          Â is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. [No text in field]
        </div>
      </div>
      <div
        class="my-s md:my-xs mb-s flex items-start md:flex-row md:items-center md:space-x-4"
      >
        <a
          aria-label="Default State"
          class="flex w-fit items-center whitespace-normal rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray disabled:hover:bg-theme-btn-bg disabled:cursor-not-allowed text-black mr-m"
          data-gtm-click=""
          data-gtm-dl-event="cta_click"
          href="/"
          role="link"
          tabindex="0"
          target=""
          title=""
        >
          Default State
          <span
            class="ml-xxs"
          >
            <svg
              fill="none"
              height="14"
              role="img"
              viewBox="0 0 14 14"
              width="14"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                New Tab Icon
              </title>
              <path
                d="M10.9999 7.66712V11.6671C10.9999 12.0207 10.8594 12.3598 10.6094 12.6099C10.3594 12.8599 10.0202 13.0004 9.6666 13.0004H2.33332C1.9797 13.0004 1.64057 12.8599 1.39052 12.6099C1.14047 12.3598 1 12.0207 1 11.6671V4.33381C1 3.98019 1.14047 3.64106 1.39052 3.39101C1.64057 3.14096 1.9797 3.00049 2.33332 3.00049H6.33329"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.2"
              />
              <path
                d="M9 1H13V4.99997"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.2"
              />
              <path
                d="M5.66699 8.33328L13.0003 1"
                stroke="white"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.2"
              />
            </svg>
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
